src/utils/MapManager.ts

```diff
@@ -0,0 +1,163 @@
+import { convertSpritesheetToIndexArray } from "@shared/canvas/CanvasUtil";
+import { Map } from "src/types/MapType";
+
+export class MapManager {
+  private _map: Map;
+
+  private tileIndexMap: number[][];
+  private spritePixelArray: Uint8Array;
+  private mapPixelArray: Uint8Array;
+
+  constructor(map: Map) {
+    if (map.width <= 0 || map.height <= 0) {
+      throw new Error("Map _width and height must be greater than 0");
+    }
+
+    if (map.mapData.length !== map.width * map.height * map.stride) {
+      throw new Error(`Map data length (${map.mapData.length}) does not match expected size (${map.width * map.height * map.stride})`);
+    }
+
+    this._map = map;
+
+    //FIXME should aready be converted in Project
+    this.spritePixelArray = convertSpritesheetToIndexArray(map.spriteSheet);
+    this.tileIndexMap = this.parseMapData();
+    this.mapPixelArray = this.buildMapPixelArray();
+  }
+
+  public getMapPixelArray(): Uint8Array {
+    return this.mapPixelArray;
+  }
+
+  public getTileAt(x: number, y: number): number | null {
```

Comment:
`number | undefined` and use `undefined` over `null` please.
We should maybe define a `Maybe` type.